AlfrescoApp/Model/Managers/RealmSyncManager+CoreDataMigration.m

```diff
@@ -34,11 +34,55 @@ @implementation RealmSyncManager (CoreDataMigration)
 
 - (BOOL)isCoreDataMigrationNeeded
 {
-    BOOL isMigrationNeededResult = ![[NSUserDefaults standardUserDefaults] objectForKey:kHasCoreDataMigrationOccurred];
+    BOOL isMigrationNeededResult = ![[NSUserDefaults standardUserDefaults] objectForKey:kHasCoreDataMigrationOccurred] && [self hasSyncContentToMigrate];
     
     return isMigrationNeededResult;
 }
 
+- (BOOL)hasSyncContentToMigrate
+{
+    BOOL hasSyncContentToMigrate = NO;
+    //creating a background context
+    CoreDataSyncHelper *coreDataSyncHelper = [CoreDataSyncHelper new];
+    NSManagedObjectContext *privateContext = [coreDataSyncHelper createChildManagedObjectContext];
+    
+    for(UserAccount *account in [[AccountManager sharedManager] allAccounts])
+    {
+        if(account.isSyncOn)
+        {
+            if([self hasSyncContentToMigrateInAccount:account usingContext:privateContext])
+            {
+                hasSyncContentToMigrate = YES;
```

Comment:
A break instruction could be added here. There's no point to check remaining accounts for sync content to migrate.